NetworkGameSI/Assets/Scripts/Lobby_Manager.cs

```diff
@@ -0,0 +1,214 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+using Unity.Netcode;
+using Unity.VisualScripting;
+using System;
+using UnityEngine.SceneManagement;
+
+public class Lobby_Manager : NetworkBehaviour
+{
+    //player lable improvised
+   [SerializeField] private Button _startBttn, _leaveBttn, _readyBttn;
+   [SerializeField] private GameObject _panelPrefab;
+   [SerializeField] private GameObject _ContentGO;
+   [SerializeField] private TMP_Text rdyTxt;
+
+   [SerializeField] private NetworkPlayerData _networkPlayers;
+
+   
+   private List<GameObject> _PlayerPanels = new List<GameObject>();
+   private ulong _myServerID;
+   private bool isReady = false;
+
+   private void Start()
+   {
+    _myServerID = NetworkManager.ServerClientId;
+
+    if (IsServer)
+    {
+        rdyTxt.text = "Waiting for Players";
+        _readyBttn.gameObject.SetActive(false);
+    }
+    else{
+        rdyTxt.text = "Not Ready";
+        _readyBttn.gameObject.SetActive(true);
+    }
+
+    if (_networkPlayers == null)
+        {
+            Debug.LogError("NetworkPlayerData is not assigned in the Lobby_Manager.");
+            return;
+        }
+
+    _networkPlayers._allConnectedPlayer.OnListChanged += NetPlayersChanged;
+    _leaveBttn.onClick.AddListener(LeaveBttnClick);
+    _readyBttn.onClick.AddListener(ClientRdyBttnToggle);
+   }
+
+    private void ClientRdyBttnToggle()
+    {
+        if(IsServer){return;}
+
+        isReady = !isReady;
+        if (isReady)
+        {
+            rdyTxt.text = "Ready";
+        }
+        else{
+            rdyTxt.text = "Not Ready";
+        }
+
+        RdyBttnToggleServerRpc(isReady);
+    }
+
+
+    [Rpc(SendTo.Server, RequireOwnership = false)]
+    private void RdyBttnToggleServerRpc(bool readyStatus, RpcParams rpcParams = default)
+    {
+        var senderId = rpcParams.Receive.SenderClientId;
+        Debug.Log("rdybttntoggleserverRpc called by client: " + senderId);
+
+        Debug.Log(message: "From Rdy bttn RPC");
+        _networkPlayers.UpdateReadyClient(rpcParams.Receive.SenderClientId, readyStatus);
+    }
+
+    private void LeaveBttnClick()
+    {
+        if(!IsServer)
+        {
+            QuitLobbyServerRpc();
+        }
+        else{
+            foreach(PlayerInfoData playerdata in _networkPlayers._allConnectedPlayer)
+            {
+                KickUserButton(playerdata._clientId);
+            }
+        }
+        NetworkManager.Shutdown();
+    }
+
```

Comment:
NetworkManager.Shutdown(); should be moved and nested with the else statement, where all the context of the server call should be. This should fix some issues when your client hits  the leave button, but doesn't get the scene reloaded.